### ISSUE TITLE
fn.apply is not a function when addDecorator is used

### ISSUE THREAD

**Issue by tobias-kuendig** [storybookjs/storybook#8719]
I did add Storybook to a `vue-cli` project using

```
npx -p @storybook/cli sb init --type vue
```

I'm using vue-i18n, so I'm injecting it to the root instance using a decorator like so in my config.js:

```ts
const i18n = setupI18N()
addDecorator({
  i18n,
  beforeCreate: function() {
    this.$root._i18n = this.$i18n;
  },
  template: '<div><slot /></div>',
})
```

This results in the following error:

![grafik](https://user-images.githubusercontent.com/8600029/68198589-a1808600-ffbc-11e9-9fa1-3439c84be694.png)

Removing the `addDecorator` call removes this error, but obviously then the missing `i18n` property is reported.

What am I missing?

**Comment by Aaron-Pool** [comment]
@tobias-kuendig There is actually a vue cli plugin for storybook. Would you mind instead initializing storybook in your app with that, and see if the error persists? https://github.com/storybookjs/vue-cli-plugin-storybook

**Comment by tobias-kuendig** [comment]
Thank you for your response, @Aaron-Pool!

I did switch to the vue-cli plugin, didn't know about that! The problem persisted, though.

I did find out that I have used an outdated or invalid addDecorator syntax. I did change my `addDecorator` call to this and things started working:

```ts
const i18n = setupI18N('de')
addDecorator(() => ({
  template: '<story/>',
  i18n,
}))
```

Or use this method to skip the initialization of vue-i18n alltogether.

```ts
addDecorator(() => ({
  template: '<story/>',
  i18n: new VueI18n({ silentTranslationWarn: true }),
}))
```
